### PR TITLE
Refactor GitHub Action to TypeScript

### PR DESCRIPTION
**Pull request by jaraco-krystian-slowik**

### CODE REVIEW

**Comment by jaraco-krystian-slowik** [comment]
Example pass job: https://github.com/jaracogmbh/research-kotlin-angular-pw-toxiproxy/actions/runs/10656258475/job/29534711463

Example failed job: https://github.com/jaracogmbh/research-kotlin-angular-pw-toxiproxy/actions/runs/10656279572

File: src/__tests__/checkDockerInstallation.test.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  it("should attempt to install Docker if not installed", async () => {
+    mockExecCommand.mockResolvedValue("");
+
+    await checkDockerInstallation();
+
+    expect(Logger.info).toHaveBeenCalledWith(
+      "Docker is not installed. Installing Docker..."
+    );
+    expect(execCommand).toHaveBeenCalledWith(
+      "curl -fsSL https://get.docker.com -o get-docker.sh"
```
Comment (jrse) [inline src/__tests__/checkDockerInstallation.test.ts:32]: maybe put the url to the top

File: action.yml [hunk 2]
```diff
@@ -22,84 +22,5 @@ inputs:
...
-          fi
-          if [ $i -eq $max_retries ]; then
-            echo "Services did not become ready within the allocated time"
-            docker-compose -f "$compose_file" ps
-            docker-compose -f "$compose_file" logs
-            exit 1
-          fi
-        done
-      shell: bash
+  using: "node20"
```
Comment (jrse) [inline action.yml:25]: isnt there a version for node20?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  },
+  "keywords": [],
+  "author": "jaracogmbh",
+  "license": "MIT",
+  "dependencies": {
+    "@actions/core": "^1.10.1",
+    "@actions/github": "^6.0.0"
+  },
+  "devDependencies": {
+    "@types/jest": "^29.5.12",
```
Comment (jrse) [inline package.json:19]: can we use dependabot to keep them up to date

File: src/actions/checkDockerInstallation.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import { execCommand } from "../utils/execCommand";
+
+export async function checkDockerInstallation(): Promise<void> {
+  try {
+    const isInstalled = await execCommand("command -v docker");
+    if (!isInstalled) {
+      Logger.info("Docker is not installed. Installing Docker...");
+
+      await execCommand(`curl -fsSL https://get.docker.com -o get-docker.sh`);
+      await execCommand(`sh get-docker.sh`);
```
Comment (jrse) [inline src/actions/checkDockerInstallation.ts:11]: also put the command in a variable to the top

**Review by jrse — APPROVED** [review]

File: src/actions/checkDockerInstallation.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import { Logger } from "../utils/logger";
+import { execCommand } from "../utils/execCommand";
+
+export async function checkDockerInstallation(): Promise<void> {
+  try {
+    const isInstalled = await execCommand("command -v docker");
+    if (!isInstalled) {
+      Logger.info("Docker is not installed. Installing Docker...");
+
+      await execCommand(`curl -fsSL https://get.docker.com -o get-docker.sh`);
```
[the remark this answers is not in the document]
Comment (krystianslowik) [reply src/actions/checkDockerInstallation.ts]: done in https://github.com/jaracogmbh/docker-compose-health-check-action/pull/2/commits/a358106562b9ec6fa7cb73c30e87f9aec4809ec1

File: src/actions/checkDockerInstallation.ts (hunk 4 above)
Comment (krystianslowik) [reply src/actions/checkDockerInstallation.ts]: done in https://github.com/jaracogmbh/docker-compose-health-check-action/pull/2/commits/a358106562b9ec6fa7cb73c30e87f9aec4809ec1

File: package.json (hunk 3 above)
Comment (krystianslowik) [reply package.json:19]: done in https://github.com/jaracogmbh/docker-compose-health-check-action/pull/2/commits/6a16bf8d88a23c25b99ab00f50a00c17376c74b1

File: src/__tests__/checkDockerInstallation.test.ts (hunk 1 above)
Comment (krystianslowik) [reply src/__tests__/checkDockerInstallation.test.ts]: done in https://github.com/jaracogmbh/docker-compose-health-check-action/pull/2/commits/a358106562b9ec6fa7cb73c30e87f9aec4809ec1

File: action.yml (hunk 2 above)
Comment (krystianslowik) [reply action.yml]: updated to use actions/setup-node@v4
https://github.com/jaracogmbh/docker-compose-health-check-action/pull/2/commits/3160ae5d77ee55c944e53cca9c567a88e314447e

**Review by krystianslowik — APPROVED** [review]

**Pull request merged**